### DISCUSSION TITLE
Next js 13 caching

### DISCUSSION THREAD

**Discussion by sargsyanvahe** [vercel/community#3351] [Help]
### Summary

Hello everyone,

I'm experiencing an issue with caching and revalidation in Next.js 13. When I run the app locally using "dev" mode, as well as "start" mode with a production build, the caching with revalidation works perfectly. However, when I deploy the app to Vercel, the caching seems to break for a specific function.

To provide more context, I have two functions in my app. The first function makes about 10 API requests, and the caching for these requests works without any problems. However, the second function, which involves just two API requests and some additional logic, doesn't seem to work with caching on Vercel.

I would appreciate any insights or suggestions on what could be causing this problem. Has anyone else encountered a similar issue with Next.js caching on Vercel? If so, how did you resolve it?

Thank you in advance for your help!

### Steps to Reproduce

delploy to vercel

**Comment by leerob** [comment]
Could you provide a demo URL and a reproducible example? 

https://vercel.com/guides/creating-a-minimal-reproducible-example

**Reply by sargsyanvahe** [reply]
I added export const runtime = "edge"; in component where I get data and seems fixed issue, maybe that was problem?

**Comment by JuaniSilva** [comment]
I have a similar problem. I have some data that is fetched on a server component, then revalidated from the client when needed. When the client makes de request, it always does with the `cache: "no-store"` flag. 
Even tho it has `must-revalidate` in the `Cache-control` header, it always return stale data. I tried adding a million things to the configuration of the route, fetch request, forcing revalidation, using axios and more but it always returned me stale data. This happens only on production. In development it works like a charm.
![image](https://github.com/vercel/community/assets/69092286/7f0daa87-c98e-4675-bc84-5585f9d2b01b)
